caliban/platform/cloud/core.py

```diff
@@ -660,6 +661,10 @@ def submit_ml_job(
     if image_tag is None:
       image_tag = generate_image_tag(project_id, docker_args, dry_run=dry_run)
 
+    image_id = image_tag[image_tag.rindex('/')+1:image_tag.index(':')]
```

Comment:
pull this logic out into a separate named fn so we know what it's doing (I can't understand it as is)

Reply:
Done